### ISSUE TITLE
Open Semantic Search Server - Bad File URL

### ISSUE THREAD

**Issue by blion-tec** [opensemanticsearch/open-semantic-search#118]
Hello!
I've just indexed a bunch of files and realized that all the files are indexed as in the Desktop version, with file://localhost/<name_of_file> type URL, so they aren't accessible in any way.
Is there a way to correct the references so the files are accessible outside of the box, from any machine web browser?
Example:
![34527408_10156367071797246_7736711648995442688_n](https://user-images.githubusercontent.com/36870003/41131962-d1a262d8-6a83-11e8-913b-2327a17b0847.jpg)

**Comment by rafael844** [comment]
Did everything as you said, but its not working.
Im using OSS Desktop. 
Shared the folder sged  in the VM
created the link:
 ln -s /home/user/Documents/sf_sged/
and  chmod 777 /home/user/Documents/sf_sged/

the /etc/opensemanticsearch/connector-files:
config['mappings'] = { "/var/www/html/": "http://10.200.52.140/" }

the apache2.conf
Listen 10.200.52.140:80

but it not works,
It indexes, and I can see the preview and Tagging & annotation fine, but when I try to open the file it retrives:
Not Found
The requested URL /home/user/Documents/sf_sged/83/id_127_File.pdf  was not found on this server.


Whats could be wrong ?

**Comment by Olydia** [comment]
I'm having the same issue, I did everything still I can't open a file on OSS server ?
Any idea on how to fix the problem 
Thank you ! :)

**Comment by barrydegraaff** [comment]
yeah this is definitively annoying, as changing the setting config['mappings'] requires a re-index. And that takes days. As a new user installing the opensemanticsearch server package I was not aware of the fact that this used to be a desktop application. Putting file:// links in a web application is IMHO not that useful, as that means it is a single-user/single-computer web application.

It would be nice if the path override could be configured to just change the links in the web-ui, so re-indexing would not be necessary. In the mean time, I created a browser extension with some javascript that re-writes my opensemanticsearch urls.

**Comment by barrydegraaff** [comment]
In the end all I needed was:
config['mappings'] = { "/": "https://mydomain.com/" }

And I purged my entire installation and reindexed.

Since some traces are left behind after purging, I also had a background task to make sure no ETL process could start before I could copy a pre-configured config file into place, basically cp'ing  /root/connector-files  /etc/opensemanticsearch/connector-files all the time.

**Comment by ufukayyildiz** [comment]
> Did everything as you said, but its not working.
> Im using OSS Desktop.
> Shared the folder sged in the VM
> created the link:
> ln -s /home/user/Documents/sf_sged/
> and chmod 777 /home/user/Documents/sf_sged/
> 
> the /etc/opensemanticsearch/connector-files:
> config['mappings'] = { "/var/www/html/": "http://10.200.52.140/" }
> 
> the apache2.conf
> Listen 10.200.52.140:80
> 
> but it not works,
> It indexes, and I can see the preview and Tagging & annotation fine, but when I try to open the file it retrives:
> Not Found
> The requested URL /home/user/Documents/sf_sged/83/id_127_File.pdf was not found on this server.
> 
> Whats could be wrong ?

worked for me. thanks
